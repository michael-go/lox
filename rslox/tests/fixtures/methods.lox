class Scone {
  topping(first, second) {
    print "scone with " + first + " and " + second;
  }
}

var scone = Scone();
scone.topping("berries", "cream");

print "/////";

class Nested {
  method() {
    print "`this` in method:";
    print this;

    fun function() {
      print "`this` in nested function:";
      print this;
    }

    function();
  }
}

Nested().method();

print "/////";

class Counter {
  init() {
    this.count = 0;
  }

  up() {
    this.count = this.count + 1;
  }

  print_current() {
    print this.count;
  }
}

var counter = Counter();
counter.up();
counter.up();
counter.print_current();