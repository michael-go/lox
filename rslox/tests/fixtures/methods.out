# exit code: 0
# stdout:
scone with berries and cream
/////
`this` in method:
Nested instance
`this` in nested function:
Nested instance
/////
2

# stderr:

